frontend/src/components/admin/AssignedStoryTranslationsTable.tsx

```diff
@@ -91,6 +136,68 @@ const AssignedStoryTranslationsTable = ({
     setStoryTranslations(newStoryTranslations);
   };
 
+  const onConfirmAssignStory = async () => {
+    const assignUser = storyToAssign!!.storyTranslationId
+      ? assignReviewer
+      : assignTranslator;
+
+    const err = await assignUser();
+    closeAssignStoryModal(false);
+    closeConfirmAssignStoryModal();
+
+    if (err !== null) {
+      window.alert(err);
+    } else {
+      setStoryAssignStage(StoryAssignStage.SUCCESS);
+      // TODO: remove this line and add story translation locally
+      window.location.reload();
+    }
+  };
+
+  const [createTranslation] = useMutation<{
+    createStoryTranslation: CreateTranslationResponse;
+  }>(CREATE_TRANSLATION);
+  const assignTranslator = async (): Promise<string | null> => {
+    try {
+      const storyTranslationData = {
+        storyId: storyToAssign!!.storyId,
+        translatorId: userId,
+        language: storyToAssign!!.language,
+      };
+      const result = await createTranslation({
+        variables: { storyTranslationData },
+      });
+      return result.data?.createStoryTranslation.story.id
+        ? null
+        : "Error assigning user as translator!";
+    } catch (err) {
+      return typeof err === "string"
```

Comment:
this is an interesting pattern. Why not just `return err` ?

Reply:
oh I got the pattern from [here](https://github.com/uwblueprint/planet-read/blob/main/frontend/src/components/homepage/StoryCard.tsx#L73), I can change it to just `return err` tho

Reply:
i like how clean `return err as string` is